releasing.md

```diff
@@ -155,14 +155,46 @@ ks apply ${ENV} -c workflows
 Create a PR to update [jupyterhub_spawner.py](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/jupyterhub_spawner.py#L15) 
 to point to the newly built Jupyter notebook images.
 
-## Update the ksonnet configs
+## Create a release branch (if necessary)
 
-Create a PR to update the ksonnet configs to use the latest images built in the previous steps.
+If you aren't already working on a release branch (of the form `v${VER}-branch`, where `${VER}` is a major-minor version number), then create one.  Release branches serve several purposes:
 
-Presubmit tests should verify that the ksonnet configs work.
+1.  They allow a release wrangler or other developers to focus on a release without interrupting development on `master`,
+2.  they allow developers and sophisticated users to track the development of a release before a release candidate is declared, and
+3.  they simplify backporting critical bugfixes to a patchlevel release particular release stream (e.g., producing a `v0.1.1` from `v0.1-branch`), when appropriate.
 
-Submit the PR.
+### Release branching policy
 
-Wait for a passing postsubmit of the submitted PR.
+A release branch should be substantially _feature complete_ with respect to the intended release.  Code that is committed to `master` may be merged or cherry-picked on to a release branch, but code that is directly committed to the release branch should be solely applicable to that release (and should not be committed back to master).
 
-Tag the commit corresponding to the passing postsubmit with the appropriate tags for the release.
+## Updating the release branch and tagging a release
+
+0. Ensure that the release notes are up to date.
+1. Update the ksonnet configs on the release branch to use the latest images built in the previous steps.
+2. Presubmit tests should verify that the ksonnet configs work.
+3. Submit a PR to the release branch.
+4. Wait for a passing postsubmit of the submitted PR.  
+5. Once it passes, update this document on the release branch with specific image hashes.
+6. If no known blocker issues remain, tag a release candidate. 
+
+### Tagging a release candidate
+
+A release candidate is a tag of the form `v${VER}.${PATCHLEVEL}-RC${N}`, where `N` is a small integer, and a release candidate tag always points to a commit on the corresponding minor release branch.  Push this tag to GitHub and announce a release vote on kubeflow-discuss.
```

Comment:
We should verify that postsubmits pass by finding the prow job
https://prow.k8s.io/?repo=kubeflow%2Fkubeflow&job=kubeflow-postsubmit

It might be more obvious to do
v${MAJOR}.${MINOR}.${PATCHLEVEL}-RC${N}

Reply:
Lets do RC.${N} to match K8s
https://github.com/kubernetes/kubernetes/releases/tag/v1.10.0-rc.1
https://github.com/kubernetes/kubernetes/releases?after=v1.10.0-beta.2

And this allows the tag (RC/alpha/beta) to be easily split from the iteration.

Reply:
Sounds good.  I'll make the changes.